bmds_server/common/auth.py

```diff
@@ -0,0 +1,31 @@
+from rest_framework import authentication, permissions
+
+
+class SessionCsrfAuthentication(authentication.SessionAuthentication):
+    """
+    Custom Session auth where CSRF is enforced for anonymous requests.
+    """
+
+    def authenticate(self, request):
```

Comment:
We override the default drf SesssionAuthentication method, enforce csrf whenever a user exists OR the request method is not safe (ie., not GET). This is the original method:

```python
def authenticate(self, request):
        """
        Returns a `User` if the request session currently has a logged in user.
        Otherwise returns `None`.
        """

        # Get the session-based user from the underlying HttpRequest object
        user = getattr(request._request, 'user', None)

        # Unauthenticated, CSRF validation not required
        if not user or not user.is_active:
            return None

        self.enforce_csrf(request)

        # CSRF passed with authenticated user
        return (user, None)
```